orangecontrib/snom/preprocess/utils.py

```diff
@@ -57,20 +57,38 @@ def _image_from_table(data, image_opts):
     )
     return hypercube[:, :, 0], indices
 
-
+# Introdueced run_all optional argument to run for all attributes 
+# in the dataset (same for PreprocessImageOpts2DOnlyWholeReference)
 class PreprocessImageOpts2DOnlyWhole(PreprocessImageOpts):
-    def __call__(self, data, image_opts):
-        data = _prepare_table_for_image(data, image_opts)
-        try:
-            image, indices = _image_from_table(data, image_opts)
-            transformed = self.transform_image(image, data)
-            col = transformed[indices].reshape(-1)
-        except InvalidAxisException:
-            col = np.full(len(data), np.nan)
-        if len(data):
-            with data.unlocked(data.X):
-                data.X[:, 0] = col
-        return data
+    def __call__(self, data, image_opts, run_all=False):
+        if run_all:
+            attrs_to_run = [v.name for v in data.domain.attributes]
+            newdata = data.copy()
+        else:
+            # This is only for the preview for a single feature image
+            # So it only processes one image
+            attrs_to_run = [image_opts["attr_value"]]
+            newdata = _prepare_table_for_image(data, image_opts)
+
+        M = np.full(np.shape(newdata.X), np.nan, dtype="float")
+        for i, attr in enumerate(attrs_to_run):
+            image_opts["attr_value"] = attr
+            try:
+                temp = _prepare_table_for_image(newdata, image_opts)
```

Comment:
Why is this called when it is unused? This needs a nicer way to raise the exception. (And probably the thing is not checked here.

Reply:
I only meant to say that the output, `temp`, is unused, so therefore this function call is only used for raising possible exceptions, so it looks strange.